virttest/qemu_vm.py

```diff
@@ -1650,9 +1651,9 @@ def sort_key(dev):
 
         # Some versions of windows don't support more than 2 sockets of cpu,
         # here is a workaround to make all windows use only 2 sockets.
-        if (vcpu_sockets and vcpu_sockets > 2 and
-                params.get("os_type") == 'windows'):
-            vcpu_sockets = 2
+        if (vcpu_sockets and win_max_vcpu_sockets and
```

Comment:
Here you removed the `os_type` check, but also you set `win_max_vcpu_sockets ` and its default is 2, so for Linux guests, they will also go to this step.

Reply:
Okay,Thanks for your review.So we still need to check the os type now.

Reply:
Restore the os_type check is okay for me, or leave `win_max_vcpu_sockets` empty for other OS, this should be the correct logic of the code. Anyway, this is up to you.

Reply:
Ok thanks @PaulYuuu.
I would like to keep the current code changes, your suggestion is reasonable, but I think it can be done in the future. Thanks!